### ISSUE TITLE
Coroutines Context Cancellation issue

### ISSUE THREAD

**Issue by halimpuckjava** [Kotlin/kotlinx.coroutines#1156]
hi
```
fun main() = runBlocking {
    calculSum(coroutineContext) { (1..1_000_000_000).sum() }
    println("done")
}

fun calculSum(parentCtx: CoroutineContext, action: suspend () -> Int) {
    action.startCoroutine(object : Continuation<Int> {
        override val context: CoroutineContext get() = Job(parentCtx[Job]) + Dispatchers.Default

        override fun resumeWith(result: Result<Int>) {
            result.getOrNull()?.also {
                println("resume with sum: $it")
                context.cancel().also { println("context is: ${if(context.isActive) "active" else "cancelled"}") }
            }
        }
    })
}

in execution i see:

done
resume sum: -243309312
context is: active
```
context still active after call cancel() !! 
any explanation for this issue ?

**Comment by elizarov** [comment]
What's the purpose of this code? What were you trying to achieve?

**Comment by halimpuckjava** [comment]
just cancel context (default-dispatcher) when resume but context still active and default-dispacher still runnng !!!!

**Comment by LouisCAD** [comment]
@halimpuckjava The snippet you provided seems overly complex.
Can you please tell what you're trying to achieve so we can help further?

**Comment by halimpuckjava** [comment]
thanks

i make a regular function "calculSum" (calcul sum of 1_000_000_000) and pass a suspend lambda & context as arguments, in second part (suspend lambda) just override context of continuation with conetxt pass as argument from "calculSum" so when this suspend lambda complete it resume with the sum and cancel the context in continuation, context still active after call of cancel() !!!!

**Comment by LouisCAD** [comment]
That explains the what just like your code did, but not the why.

I have never overridden the context of a continuation even though I have used coroutines almost everyday for almost 2 years.

Also, I don't see the point of using the low-level `startCoroutine` function when using kotlinx.coroutines which provides `launch`.

I failed to understand why you want to play with the context in that way, and your almost 2 tweets (385 chars) worth sentence (!!!!) didn't really help me.

Note that kotlinx.coroutines, and Kotlin overall, are use-case oriented, not theory oriented, because use cases are easier to grasp than theory detached from it.
I hope you understand and I can help you further.
